src/test/java/com/karankumar/bookproject/backend/service/BookServiceTest.java

```diff
@@ -246,4 +253,51 @@ private Book createBookAndSetAllAttributes() {
     void deleteBooks() {
         bookService.deleteAll();
     }
+    
+    @Test
+    void bookWithoutPublisherWithAuthorSaved() {
+    	SoftAssertions softly = new SoftAssertions();
+
+        // when
+    	Book book = new Book("Book without author", author, toRead);
+    	book.setPublisher(null);
+        bookService.save(book);
+
+        // then
+        softly.assertThat(authorService.count()).isOne();
+        softly.assertThat(bookService.count()).isOne();
+        softly.assertAll();
+    }
+    
+    @Test
+    void bookWithPublisherWithAuthorSaved() {
+    	SoftAssertions softly = new SoftAssertions();
```

Comment:
In line with our style guide, assertSoftly should be used here (saves us from needing to assert all at the end)